presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -469,12 +474,21 @@ else if (!partitionColumnNames.contains(name)) {
         table.setDbName(schemaName);
         table.setTableName(tableName);
         table.setOwner(tableOwner);
-        table.setTableType(TableType.MANAGED_TABLE.toString());
+        table.setTableType(hiveTableType.toString());
         String tableComment = "Created by Presto";
         if (sampled) {
             tableComment = "Sampled table created by Presto. Only query this table from Hive if you understand how Presto implements sampling.";
         }
-        table.setParameters(ImmutableMap.of("comment", tableComment));
+        ImmutableMap.Builder<String, String> tableParameters = ImmutableMap.builder();
+        tableParameters.put("comment", tableComment);
+
+        //Setting the table type to EXTERNAL_TABLE is not enough
+        //as Hive metastore also looks for the "EXTERNAL" property
+        if (hiveTableType.equals(EXTERNAL_TABLE)) {
```

Comment:
Please see [this](https://github.com/apache/hive/blob/master/metastore/src/java/org/apache/hadoop/hive/metastore/MetaStoreUtils.java#[L1319-L1329]).